src/v/cluster/cluster_discovery.h

```diff
@@ -105,17 +135,36 @@ class cluster_discovery {
      */
     static std::optional<int32_t> get_node_index_in_seed_servers();
 
-    // Sends requests to each seed server to register the local node UUID until
-    // one succeeds. Upon success, sets `node_id` to the assigned node ID and
-    // returns true.
-    ss::future<bool> dispatch_node_uuid_registration_to_seeds(model::node_id&);
+    /**
+     * Validate cluster_bootstap_info responses from seeds for the purpose
+     * of at least being an amnesiac seed server, and check if it actually
+     * can be such.
+     *
+     * Amnesiac is defined as a cluster member that has had its local storage
+     * wiped out.
+     *
+     * Validation failures: two or more distinct cluster_uuid reported by other
+     * seeds; configuration mismatch of the local with other seeds.
+     *
+     * \return true if the server is apparently an amnesiac seed server, and is
+     * validated for that
+     * \return false if the server is rather a cluster founder than an amnesiac
+     * \throw std::runtime_error if it cannot be either of them (validation has
+     * been failed)
+     */
+    bool is_amnesiac_seed_server() const;
```

Comment:
nit: i like the name `amnesiac` but maybe it would be easier to grasp with `is_wiped_seed_server` ?